dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java

```diff
@@ -723,7 +723,7 @@ else if (DOWNLOAD_ARTIFACT_LIST.equals(key)) {
                 throw new IllegalArgumentException();
             }
         }
-        return data.get(key);
+        return data.put(key, value);
```

Comment:
What this method used to do is :
1. populate HashMap `data` after validating the key.
2. Return the value to which the specified key is now mapped, or null if this map contains no mapping for the key (the key wasn't added)

This changes the method to
1. populate HashMap `data` after validating the key.
2. If the key doesn't meet any criteria in the if block, add the key anyways. If the key did meet criteria in the if block, add it again.
3. Return the previous value associated with key, or null if there was no mapping for key.

Returning the value from `put()` is a significant change in behaviour, I don't think this is what is intended. Instead of making this change we need to add the key / value pair in the if block or in an `else` block after line 725.

Reply:
if we are only concerned about the key `action.error.message` then we should add an else if and check for that exact key: 
```
else if (key.equals("action.error.message")) {
            if (value instanceof String) {
                data.put(key, value);
            } else {
                throw new IllegalArgumentException();
            }
        }
```